modules/oe_webtools_wtag/src/Plugin/Field/FieldWidget/WtagWidget.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_webtools_wtag\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+
+/**
+ * Plugin implementation of the 'oe_webtools_wtag_autocomplete' widget.
+ *
+ * @FieldWidget(
+ *    id = "oe_webtools_wtag",
+ *    label = @Translation("Webtools Wtag"),
+ *    description = @Translation("A Webtools Wtag widget that allows the selection of SKOS Concepts in a modal window."),
+ *    field_types = {
+ *      "skos_concept_entity_reference"
+ *    }
+ *  )
+ */
+class WtagWidget extends WidgetBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'modal_title' => t('Wtag'),
+      'modal_description' => t('Use the search field or the tree view to find and add one or more tags.'),
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element['modal_title'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Modal title'),
+      '#default_value' => $this->getSetting('modal_title'),
+      '#description' => $this->t('The title of the modal.'),
+    ];
+    $element['modal_description'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Modal description'),
+      '#default_value' => $this->getSetting('modal_description'),
+      '#description' => $this->t('The introductory text on the modal.'),
+    ];
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+
+    $summary[] = $this->t('Modal title: @title', ['@title' => $this->getSetting('modal_title')]);
+    $summary[] = $this->t('Modal description: @description', ['@description' => $this->getSetting('modal_description')]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function isApplicable(FieldDefinitionInterface $field_definition) {
+    // This widget is only applicable to unlimited cardinality SKOS Concept
+    // reference fields that reference the DET vocabulary.
+    if ($field_definition->getFieldStorageDefinition()->getCardinality() !== FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED) {
+      return FALSE;
+    }
+
+    $concept_schemes = $field_definition->getSetting('handler_settings')['concept_schemes'];
+    if (!in_array('http://data.europa.eu/uxp/det', $concept_schemes)) {
```

Comment:
Should we allow to use this widget for a field that is configured to use the DET vocabulary AND some other vocabulary? It seems weird as it essentially prevents the user from choosing anything but the det vocabulary

Reply:
The Webtools Wtag widget only allows DET, so I don't think the user would be able to choose terms from other vocabularies.

Reply:
@imanoleguskiza They only built support for DET, quite hardcoded on their end as well..

Reply:
@imanoleguskiza Yes, true.
@22Alexandra could you please update this to ensure that there is a single concept scheme selected and that it is DET? Thanks